### PR TITLE
lib/posix-process: remove clone3 syscall

### PR DESCRIPTION
**Pull request by John-Ted**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### Description of changes

As clone3 currently does not work, and glibc falls back to clone if clone3 is not available (tested on 2.35), the syscall can be removed until a working implementation is available.

Log section on example program https://github.com/unikraft/static-pie-apps/pull/91:
```
[    0.342066] dbg:  [libsyscall_shim] <uk_syscall_binary.c @   88> Binary system call request "clone3" (435) at ip:0x40018696d (arg0=0x40006fa10, arg1=0x58, ...)
[    0.344056] dbg:  [libsyscall_shim] <uk_syscall6_r.c @  836> syscall "clone3" is not available
clone3(0x40006fa10, 0x58, ...) = Function not implemented (-38)
[    0.346030] dbg:  [libsyscall_shim] <uk_syscall_binary.c @   88> Binary system call request "clone" (56) at ip:0x400186926 (arg0=0x3d0f00, arg1=0x1000012270, ...)
[    0.348068] dbg:  [libposix_process] <clone.c @  491> (int) uk_syscall_r_clone((unsigned long) 0x3d0f00, (void *) 0x1000012270, (int *) 0x1000012910, (int *) 0x1000012910, (unsigned long) 0x1000012640)
[    0.350561] dbg:  [libposix_process] <clone.c @  350> uk_syscall_r_clone(
[    0.351498] dbg:  [libposix_process] <clone.c @  351>  flags: 0x3d0f00 [ VM FS FILES SIGHAND THREAD SYSVSEM SETTLS PARENT_SETTID CHILD_CLEARTID ]
[    0.353313] dbg:  [libposix_process] <clone.c @  381>  parent_tid: 0x1000012910
[    0.354329] dbg:  [libposix_process] <clone.c @  383>  child_tid: 0x1000012910
[    0.355351] dbg:  [libposix_process] <clone.c @  384>  stack: 0x1000012270
[    0.356302] dbg:  [libposix_process] <clone.c @  385>  tls: 0x1000012640
[    0.357228] dbg:  [libposix_process] <clone.c @  386>  <return>: 0x400186926
[    0.358206] dbg:  [libposix_process] <clone.c @  387> )
[    0.358930] dbg:  [libposix_process] <clone.c @  418> Allocating an Unikraft TLS for the new child, parent called from context with custom TLS
[    0.360713] dbg:  [libuksched] <thread.c @  254> uk_thread 0x5fec2b018 (elfapp): ctx:0x5fec2b018, ectx:0x5fec2c1a0, tlsp:0x5fec2c190
[    0.362388] dbg:  [libcontext] <tls.c @  173> tls_area_init: target: 0x5fec2c020 (376 bytes)
[    0.363561] dbg:  [libcontext] <tls.c @  175> tls_area_init: copy (.tdata): 4 bytes
[    0.364629] dbg:  [libcontext] <tls.c @  177> tls_area_init: uninitialized (.tbss): 364 bytes
[    0.365816] dbg:  [libcontext] <tls.c @  179> tls_area_init: pad: 0 bytes
[    0.366755] dbg:  [libcontext] <tls.c @  181> tls_area_init: tcb: 8 bytes
[    0.367696] dbg:  [libcontext] <tls.c @  183> tls_area_init: tcb self ptr: 0x5fec2c190
[    0.368812] dbg:  [libcontext] <tls.c @  204> (tls_area): 5fec2c020  0d f0 b0 b0 00 00 00 00  00 00 00 00 00 00 00 00  |................|
[    0.370534] dbg:  [libcontext] <tls.c @  204> (tls_area): 5fec2c030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
[    0.372263] dbg:  [libcontext] <tls.c @  204> *
[    0.372894] dbg:  [libcontext] <tls.c @  204> (tls_area): 5fec2c190  90 c1 c2 fe 05 00 00 00                           |........|
[    0.374528] dbg:  [libuksched] <thread.c @  144> uk_thread 0x5fec2b018 (elfapp) init: Call initialization 0x122f50()...
[    0.376053] dbg:  [libposix_process] <process.c @  173> Process PID 0: New thread TID 1
[    0.377168] dbg:  [libposix_process] <process.c @  430> thread 0x5fec2b018 (elfapp): New thread with TID: 1 (PID: 0)
[    0.378635] dbg:  [libposix_process] <clone.c @  451> Child is going to wake up with TLS pointer set to: 0x1000012640 (custom TLS)
[    0.380277] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x122d10() [flags: 0x10000]...
[    0.382071] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x123450() [flags: 0x100000]...
[    0.383900] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x1247f0() [flags: 0x100000800]...
[    0.385759] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x124b40() [flags: 0x400000]...
[    0.387566] dbg:  [libposix_process] <clone.c @  557> Ignoring historical CLONE_DETACHED
[    0.388697] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x124850() [flags: 0x100]...
[    0.390481] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x126310() [flags: 0x200000]...
[    0.392290] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x147d10() [flags: 0x40000]...
[    0.394084] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x150ea0() [flags: 0x400]...
[    0.395851] dbg:  [libposix_process] <clone.c @  176> posix_clone 0x5fec2b018 (elfapp) init: Call initialization 0x154d10() [flags: 0x200]...
[    0.397634] dbg:  [libposix_process] <clone.c @  461> Thread cloned 0x40000b018 (elfapp) -> 0x5fec2b018 (elfapp): 0
[    0.399088] dbg:  [libcontext] <ctx.c @   69> ukarch_ctx 0x5fec2b018: start:0x400186926 sp:0x1000012270
clone(CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, va:0x1000012270, <ref:0x1000012910>pid:1, <ref:0x1000012910>pid:1, va:0x1000012640) = pid:1
```

### CODE REVIEW

**Review by flpostolache — APPROVED** [review]
Hello, @John-Ted! Thank you for submitting this PR. I've tested it and it also works on my side.

Reviewed-by: Florin Postolache <florin.postolache.of@gmail.com>

File: lib/posix-process/Makefile.uk [hunk 1]
```diff
@@ -17,7 +17,7 @@ LIBPOSIX_PROCESS_SRCS-$(CONFIG_LIBPOSIX_PROCESS_CLONE) += $(LIBPOSIX_PROCESS_BAS
 COMPFLAGS-$(CONFIG_LIBPOSIX_PROCESS_PIDS) += -fno-builtin-exit -fno-builtin-exit-group
 
 UK_PROVIDED_SYSCALLS-$(CONFIG_LIBPOSIX_PROCESS_CLONE) += clone-5
-UK_PROVIDED_SYSCALLS-$(CONFIG_LIBPOSIX_PROCESS_CLONE) += clone3-2
+#UK_PROVIDED_SYSCALLS-$(CONFIG_LIBPOSIX_PROCESS_CLONE) += clone3-2
```
Comment (skuenzer) [inline lib/posix-process/Makefile.uk]: I'd remove this line here instead of commenting out.

**Review by skuenzer — CHANGES_REQUESTED** [review]
HI @John-Ted,

thanks for this workaround. Can you remove the clone3 system call registration line in `Makefile.uk` instead of commenting out. Then we are good to go with this PR.

Thanks,

Simon

**Comment by skuenzer** [comment]
Hi @John-Ted, can you merge the two commits you have now into one single commit? Thanks

**Review by skuenzer — APPROVED** [review]
Reviewed-by: Simon Kuenzer <simon@unikraft.io>
Approved-by: Simon Kuenzer <simon@unikraft.io>

**Pull request closed without merging**